src/plugins/data/public/query/query_string/language_service/types.ts

```diff
@@ -5,26 +5,22 @@
 
 import { ISearchInterceptor } from '../../../search';
 import { Query } from '../../../../public';
+import { EditorInstance } from '../../../ui/query_editor/editors';
 
 export interface LanguageConfig {
   id: string;
   title: string;
   search: ISearchInterceptor;
   getQueryString: (query: Query) => string;
-  searchBar?: {
-    showQueryInput?: boolean;
-    showFilterBar?: boolean;
-    showDatePicker?: boolean;
-    showAutoRefreshOnly?: boolean;
-    dateRange?: {
-      initialFrom?: string;
-      initialTo?: string;
-    };
-  };
+  editor: (
+    collapsedProps: any,
+    expandedProps: any,
+    bodyProps: any
+  ) => EditorInstance<any, any, any>;
```

Comment:
nit: i think this could be optional. if they don't pass in an editor we use the default editor

Reply:
fast follow: export the default Editor, or required components